src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -376,10 +376,20 @@ public bool TryGetVariableType(string name, out FormulaType type)
 
             return false;
         }
-
+        
         public void UpdateSupportedFunctions(SymbolTable s)
         {
             SupportedFunctions = s;
         }
+        
+        public void AddNamedFormulas(string script, CultureInfo parseCulture = null)

```

Comment:
Consider instead having this be a general AddUserDefinitions, that adds both named formulas like this as well as any UDFs that might be in that script. At least as it stands, it's fine for them to be defined in the same "file"/string, so having separate APIs seems like a place we could trip up.